docs/docs/dev_docs/contracts/syntax/functions.md

```diff
@@ -2,53 +2,233 @@
+- [`notes`](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/oracle/notes.nr) - Provides a lot of functions related to notes, such as fetches notes from storage etc, used behind the scenes for value notes and other pre-build note implementations.
+
+
+---
+
+## Calling functions from other functions
+
+### Private -> Private
+
+In Aztec Private to Private function calls are handled by the [private kernel circuit](../../../concepts/advanced/circuits/kernels/private_kernel.md), and take place on the user's device.
+Behind the scenes, the `Aztec RPC Server` (the beating heart of Aztec that runs in your wallet) will execute all of the functions in the desired order "simulating" them in sequence. For example, a very common use-case of Private to Private interaction is calling another private function from an `account contract` (Account contracts are a general concept, more information about them can be found [here](../../wallets/writing_an_account_contract.md)).
+
+Take, for example, the following call stack:
+```
+AccountContract::entrypoint
+    |-> Foo::example_call
+        | -> Bar::nested_call
+    |-> Baz::example_call
+```
+
+In the example above the Account Contract has been instructed to call two external functions. In the first function all, to `Foo::example_call` a further nested call is performed to `Bar::nested_call`. Finally the account contract makes one last call to `Baz::example_call`.
+
+Lets further illustrate what these examples could look like
+<!-- TODO: should these move into the docs examples -->
+```rust
+// Foo contains a singular function that returns the result of Bar::nested_call
+contract Foo {
+    #[aztec(private)]
+    fn example_call(input: Field) -> pub Field {
+        Bar::at(<bar_address>).nested_call(input)
+    }
+}
 
-### `internal`
+// Bar contains a singular function that returns a `input + 1`
+contract Bar {
+    #[aztec(private)]
+    fn nested_call(input: Field) -> pub Field {
+        input + 1
+    }
+}
 
-Similar to Solidity, internal functions and vars can be accessed within the contract itself. While technically callable from other contracts, there is a dynamic check that validates the caller to ensure it's the same contract.
+// Baz contains a singular function that simply returns `10`
+contract Baz {
+    #[aztec(private)]
+    fn example_call() -> pub Field {
+        10
+    }
+}
+```
 
-### `external`
+When simulating the following call stack, we can execution flow to continue procedurally. The simulator will begin at the account contract's entry point, find a call to `Foo::example_call`, then begin to execute the code there. When the simulator executes the code in contract `Foo`, it will find the further nested call to contract `Bar::nested_call`. It will execute the code in Bar, bringing the return value back to contract `Foo`.
```

Comment:
```suggestion
When simulating the following call stack, we can expect execution flow to continue procedurally. The simulator will begin at the account contract's entry point, find a call to `Foo::example_call`, then begin to execute the code there. When the simulator executes the code in contract `Foo`, it will find the further nested call to contract `Bar::nested_call`. It will execute the code in `Bar`, bringing the return value back to contract `Foo`.
```